### PR TITLE
Meter additional container length as a function of original size

### PR DESCRIPTION
**Pull request by dsainati1**
Arrays and dictionaries, being backed by atrees underneath, do not consume a linear amount of memory as they grow. Rather, the size of these structures increases faster and faster the larger they become, as more nodes need to be added at higher levels of the tree as the number of leaves increases. 

Based on discussion with @ramtinms, we can use `log(Count())` of the underlying tree to (over)estimate the amount of overhead necessary to grow the tree. 

______

- [ ] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

File: runtime/common/metering.go
Comment (robert-e-davidson3) [inline runtime/common/metering.go:166]: Duplicate
```suggestion
```

**Review by robert-e-davidson3 — APPROVED** [review]

**Comment by turbolent** [comment]
@dsainati1 let's please get at least one review from a code owner before merging in PRs

**Comment by dsainati1** [comment]
Why did it even allow me to merge it at all? I thought a Code owner review was required to merge in the first place

File: runtime/common/metering.go
Comment (turbolent) [inline runtime/common/metering.go:157]: The code here is a duplicate of the above, maybe factor the logic out. We might forget to update one of the functions

File: runtime/common/metering.go
Comment (turbolent) [inline runtime/common/metering.go:159]: Does this assume `additionalSize` is always 1? If so, maybe remove the parameter and simplify to `newAmount = 1`. 

If not: What happens if `originalSize <= 1` and `additionalSize > 1`?  Shouldn't this still calculate the new number of intermediate nodes allocated (which won't be the same as the number of new elements)?

File: runtime/common/metering.go
Comment (turbolent) [inline runtime/common/metering.go:162]: Followed up on the formula here, not quite sure log2 is a correct over-estimation for the atree data structures: 
https://axiomzen.slack.com/archives/C0355LG8V6J/p1649889273942269?thread_ts=1649706053.685719&cid=C0355LG8V6J

**Comment by turbolent** [comment]
@dsainati1 Because you're merging into a non-`master` branch, only `master` has a code owner requirement

**Comment by turbolent** [comment]
Added a branch protection rule for `feature/*`, same as `master`

**Pull request merged**
